sdk/keyvault/Azure.Security.KeyVault.Administration/src/AccessControlClient.cs

```diff
@@ -0,0 +1,322 @@
+        /// <value></value>
+        public virtual Uri VaultUri { get; }
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="AccessControlClient"/> class for mocking.
+        /// </summary>
+        protected AccessControlClient()
+        { }
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="AccessControlClient"/> class for the specified vault.
+        /// </summary>
+        /// <param name="vaultUri">A <see cref="Uri"/> to the vault on which the client operates. Appears as "DNS Name" in the Azure portal.</param>
+        /// <param name="credential">A <see cref="TokenCredential"/> used to authenticate requests to the vault, such as DefaultAzureCredential.</param>
+        /// <exception cref="ArgumentNullException"><paramref name="vaultUri"/> or <paramref name="credential"/> is null.</exception>
+        public AccessControlClient(Uri vaultUri, TokenCredential credential)
+            : this(vaultUri, credential, null)
+        {
+
+        }
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="AccessControlClient"/> class for the specified vault.
+        /// </summary>
+        /// <param name="vaultUri">A <see cref="Uri"/> to the vault on which the client operates. Appears as "DNS Name" in the Azure portal.</param>
+        /// <param name="credential">A <see cref="TokenCredential"/> used to authenticate requests to the vault, such as DefaultAzureCredential.</param>
+        /// <param name="options"><see cref="AccessControlClientOptions"/> that allow to configure the management of the request sent to Key Vault.</param>
+        /// <exception cref="ArgumentNullException"><paramref name="vaultUri"/> or <paramref name="credential"/> is null.</exception>
+        public AccessControlClient(Uri vaultUri, TokenCredential credential, AccessControlClientOptions options)
+        {
+            VaultUri = vaultUri;
+            Argument.AssertNotNull(vaultUri, nameof(vaultUri));
+            Argument.AssertNotNull(credential, nameof(credential));
+
+            options ??= new AccessControlClientOptions();
+            string apiVersion = options.GetVersionString();
+
+            HttpPipeline pipeline = HttpPipelineBuilder.Build(options,
+                    new ChallengeBasedAuthenticationPolicy(credential));
+
+            var diagnostics = new ClientDiagnostics(options);
+            _definitionsRestClient = new RoleDefinitionsRestClient(diagnostics, pipeline, apiVersion);
+            _assignmentsRestClient = new RoleAssignmentsRestClient(diagnostics, pipeline, apiVersion);
+        }
+
+        /// <summary>
+        /// Get all role definitions that are applicable at scope and above.
+        /// </summary>
+        /// <param name="scope"> The scope of the role definition. </param>
+        /// <param name="cancellationToken"> The cancellation token to use. </param>
+        /// <returns></returns>
+        [ForwardsClientCalls]
```

Comment:
This is a pattern that will be deprecated. Instead, you need to have all the client scopes. I believe @pakrym has a Roslyn analyzer that can fix this, which he used when he made some refactoring changes that affected Search (see sdk\search\Azure.Search.Documents for an example).

Reply:
Thanks - @pakrym pointed me to it [here](https://camo.githubusercontent.com/617491a34a14d7835de9bc010058be1dd0b7e563/68747470733a2f2f692e696d6775722e636f6d2f345348774f324d2e676966). Note it won't show up unless there is a filed of type `ClientDiagnostics` defined in the client.